File: Backend_Dotnet/dotnetBackend/Models/Photographer.cs
using System.ComponentModel.DataAnnotations;
using System.ComponentModel.DataAnnotations.Schema;

namespace PhotoHub.Api.Models
{
    [Table("photographers")]
    public class Photographer
    {
        [Key]
        [Column("photographer_id")]
        public long Id { get; set; }

        [Required]
        [MaxLength(255)]
        public string Name { get; set; } = string.Empty;

        [Required]
        [MaxLength(255)]
        public string Email { get; set; } = string.Empty;

        [MaxLength(50)]
        public string? Phone { get; set; }

        [MaxLength(255)]
        public string? Location { get; set; }

        [MaxLength(100)]
        public string? Category { get; set; }

        [MaxLength(2000)]
        public string? Bio { get; set; }

        [Column("profile_picture")]
        public string? ProfileImage { get; set; }

        [Column("price_per_hour", TypeName = "decimal(10, 2)")]
        public decimal StartingPrice { get; set; }

        public double Rating { get; set; } = 5.0;

        public int ReviewCount { get; set; } = 0;

        [Column("is_verified")]
        public bool Verified { get; set; } = true;

        public bool Available { get; set; } = true;

        [Column("created_at")]
        public DateTime CreatedAt { get; set; } = DateTime.UtcNow;
    }
}

File: Backed_Java/backend_java/src/main/java/com/photohub/model/Photographer.java
package com.photohub.model;

import jakarta.persistence.*;
import lombok.*;
import org.hibernate.annotations.CreationTimestamp;
import java.time.LocalDateTime;

@Entity
@Table(name = "photographers")
@Data
@NoArgsConstructor
@AllArgsConstructor
@Builder
public class Photographer {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    @Column(name = "photographer_id")
    private Long photographerId;

    @Column(nullable = false, length = 100)
    private String name;

    @Column(nullable = false, unique = true, length = 100)
    private String email;

    @Column(nullable = false)
    private String password;

    @Column(length = 20)
    private String phone;

    @Enumerated(EnumType.STRING)
    @Column(nullable = false)
    @Builder.Default
    private Role role = Role.PHOTOGRAPHER;

    @Column(columnDefinition = "TEXT")
    private String profilePicture;

    @Builder.Default
    private String status = "active";

    private Integer experience;

    @Column(columnDefinition = "TEXT")
    private String bio;

    private String location;

    private Double rating;

    @Column(name = "price_per_hour")
    private Double pricePerHour;

    @Column(name = "is_verified")
    @Builder.Default
    private boolean isVerified = false;

    @Column(name = "cover_image", columnDefinition = "TEXT")
    private String coverImage;

    private String specialties; // Stored as comma-separated values

    @Column(name = "is_deleted")
    @Builder.Default
    private boolean isDeleted = false;

    @Column(name = "deleted_at")
    private LocalDateTime deletedAt;

    @CreationTimestamp
    @Column(name = "created_at", updatable = false)
    private LocalDateTime createdAt;
}
